include/appfwk/NamedObject.hpp

```diff
@@ -36,10 +37,36 @@ class NamedObject
    * @brief Get the name of this NamedObejct
    * @return The name of this NamedObject
    */
-  const std::string& get_name() const { return name_; }
+  virtual const std::string& get_name() const = 0;
```

Comment:
Since it is virtual I would change the return type to string. As the memory allocation is necessary to be available